src/modelfree/scheduling.py

```diff
@@ -84,3 +132,61 @@ def get_value(self, frac_remaining):
         else:
             anneal_progress = min(1.0, (1 - frac_remaining) / self.end_frac)
         return (1 - anneal_progress) * self.start_val + anneal_progress * self.end_val
+
+
+class ConditionalAnnealer(Annealer):
+    """Anneal the value depending on some condition."""
+    def __init__(self, start_val, end_val, decay_factor, thresh, window_size,
+                 min_wait, max_wait, operator, metric, get_logs):
+        super().__init__(start_val, end_val, get_logs)
+        self.decay_factor = decay_factor
+        self.thresh = thresh
+        self.window_size = window_size
+        self.min_wait = min_wait
+        self.max_wait = max_wait
+        self.operator = operator
+        self.metric = metric
+
+        self.current_param_val = start_val
+        self.last_total_episodes = 0
+
+    @classmethod
+    def from_dict(cls, cond_config, get_logs=None):
+        # provided to help keep track of arguments
+        constructor_config = {
+            'decay_factor': 0.98,
+            'thresh': 0,
+            'start_val': 1,
+            'end_val': 0,
+            'window_size': 1,
+            'min_wait': 1,
+            'max_wait': 10000,
+            'operator': operator.gt,
+            'metric': 'sparse',  # dense, length
+        }
+        if 'operator' in cond_config:
+            cond_config['operator'] = getattr(operator, cond_config['operator'])
+
+        trimmed_config = {k: v for k, v in cond_config.items() if k in constructor_config}
+        constructor_config.update(trimmed_config)
+        return cls(**constructor_config, get_logs=get_logs)
+
+    def get_value(self, frac_remaining):
+        if self.get_logs is None:
+            raise ValueError("ConditionalAnnealer requires a get_logs function to be set")
+
+        current_data = self.get_logs()
+        if current_data is None:  # if we have zero episodes thus far
+            return self.current_param_val
+        # this is fine since we are doing deque.appendleft to put together these values
+        current_wait = current_data['total_episodes'] - self.last_total_episodes
+        if current_wait < self.min_wait:
+            return self.current_param_val
+
+        metric_data = current_data[self.metric]
+        current_metric_data = itertools.islice(metric_data, self.window_size)
+        avg_metric_val = float(sum(current_metric_data)) / min(self.window_size, len(metric_data))
```

Comment:
Why not just divide by the length of `current_metric_data`?

Reply:
the object returned by `itertools.islice` doesn't have a `__len__`